### PR TITLE
chore: auto publish to npm, closes #106

### LINKED ISSUE #106: webflow-utils npm package fails to update via lerna version
### [READ] Step 1: Are you in the right place?

Yes

### [REQUIRED] Step 2: Describe your configuration

- Extension name: **/webflow-utils** 
- Extension version: **0.04**

#### Steps to reproduce:

1. Execute `npx lerna version` and select patch update for all packages.
2. Apply changes
3. All package releases, including `webflow-utils` are created and available in GitHub releases.
  - https://github.com/simplycubed/extensions/releases/tag/%40simplycubed%2Fwebflow-utils-v0.0.4
4. GitHub fails to deploy a new version of webflow-utils to npm
  - https://www.npmjs.com/package/@simplycubed/webflow-utils?activeTab=versions

##### Expected result

1. An updated version of `@simplycubed/webflow-utils@0.0.4` is available in npm

##### Actual result

GitHub Action - Failed Job - https://github.com/simplycubed/extensions/actions/runs/3972943962/jobs/6811306605

@simplycubed/webflow-utils@0.0.4 is available in GitHub Releases but not available in npm.

image

## Notes

We should consider upgrading our npm release process to use GitHub's publishing flow that gets triggered when new Releases are created.

https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages

### PR DESCRIPTION
**Pull request by anasmuhmd**

### CODE REVIEW

**Review by charlesgreen — CHANGES_REQUESTED** [review]
@anasmuhmd this will push a new release to npm even if there are no changes to webflow-utils. 

I think we should change the approach to publish to npm when a new GitHub Release is created.  

https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry

**Comment by anasmuhmd** [comment]
> @anasmuhmd this will push a new release to npm even if there are no changes to webflow-utils.

@charlesgreen The trigger will only publish when if there is a version (Logic is inside scripts/npm_publish.sh)

> I think we should change the approach to publish to npm when a new GitHub Release is created.
> 
> https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry

@charlesgreen sounds ok, but we would be triggering publish action unnecessarily on every release. Because every release doesn't trigger a publish to npm. I'm not sure if there is way to check if release is for webflow-utils package.

**Comment by charlesgreen** [comment]
Hi @anasmuhmd  Thanks for the replies. It took some time to look at the options. 
I agree that if we only use Github Actions it is difficult to know which packages are published, etc.

We can use the Lerna binary to drive the version updates, create GitHub Releases for each package, and publish the public packages to npm (Lerna knows only to publish packages set to public in the package.json) to npm. 

The open question is if we connect it to GitHub Actions or not to make it a manual or automatic process. 
I found an article with a code repo including the implementation using GitHub Actions.

- https://dev.to/xcanchal/automatic-versioning-in-a-lerna-monorepo-using-github-actions-4hij
- https://github.com/xcanchal/monorepo

There is a small change as they also support a development branch (beta release) and a main branch (production release) which I think might be a good idea to follow as well. 

Please have a look and let me know your thoughts.

**Comment by anasmuhmd** [comment]
> * https://github.com/xcanchal/monorepo

https://dev.to/xcanchal/automatic-versioning-in-a-lerna-monorepo-using-github-actions-4hij looks good to me. we can use the same approach.

**Comment by charlesgreen** [comment]
@anasmuhmd I agree. Please make the needed changes to follow the monorepo approach.

**Comment by charlesgreen** [comment]
@anasmuhmd, the mono repo also uses a main and dev branch to publish a final and beta release. 
I'm thinking we might also want to do that. What do you think?

**Comment by charlesgreen** [comment]
@anasmuhmd I'm not sure the current status of this ticket, given the new approach. Please let me know.

**Comment by anasmuhmd** [comment]
@charlesgreen I haven't updated yet.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: "Use NodeJS 14"
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14"
```
Comment (charlesgreen) [inline .github/workflows/publish.yml:24]: Update to use `actions/setup-node@v3` with Node 16

https://github.com/actions/setup-node

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      - development
+      - main
+
+jobs:
+  publish:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Checkout"
```
Comment (charlesgreen) [inline .github/workflows/publish.yml:16]: Update to use `actions/checkout@v3` with Node 16

https://github.com/actions/setup-node

**Review by charlesgreen — CHANGES_REQUESTED** [review]
@anasmuhmd I don't see it in the code but guessing that the flow also updates the tags on the GitHub repo and creates a release there as well.

My understanding is that lerna respects the package.json files so that only the webflow-utils repo will be published to NPM. 

Please let me know if my understanding is incorrect.

**Comment by anasmuhmd** [comment]
> @anasmuhmd I don't see it in the code but guessing that the flow also updates the tags on the GitHub repo and creates a release there as well.

For github release, we need to add the option '--create-release github'. I tried testing this in a different repo, I couldn't successfully do it.

> My understanding is that lerna respects the package.json files so that only the webflow-utils repo will be published to NPM.

Yes, I think so. I haven't tested it.

**Comment by anasmuhmd** [comment]
> @anasmuhmd I don't see it in the code but guessing that the flow also updates the tags on the GitHub repo and creates a release there as well.
> 
> My understanding is that lerna respects the package.json files so that only the webflow-utils repo will be published to NPM.
> 
> Please let me know if my understanding is incorrect.

Create Release option is added as well. Yes, only private packages are published.

**Review by charlesgreen — APPROVED** [review]

**Pull request merged**